src/mod/l4_thruster.c

```diff
@@ -541,27 +741,1590 @@ void GeneralReadRequest(int argc, char *argv[]){
+				waiting_between_logging = (int)( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage)/1000 );
+
+				sprintf(print_str, "\n Waiting %d s \n",waiting_between_logging  );
+				len = strlen(print_str);
+				deb_print_pure_debug((uint8_t *)print_str, len);
+
+
+
+
+				//temp_argv[0]= "6";
+				//temp_argv[1]= "99"; // Reservoir Temperature
+				//GeneralReadRequest(2, temp_argv);
+
+				THR_HARDCODED_SEQUENCES[procedure_id].substage_index++; // increase substage to procede with READ request
+				THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage = (uint32_t)timGetSystime(); // save timestamp of substage completion
+				break;
+
+					}
+
+
+
+		if ( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage) < duration ){
+				// do nothing
+			}
+		else{
+				THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage = (uint32_t)timGetSystime(); //save execution finish time of wait stage
+
+				sprintf(print_str, "\nWait and Monitor Stage = %d Complete t = %d\n", THR_HARDCODED_SEQUENCES[procedure_id].execution_index,now_timestamp);
+				len = strlen(print_str);
+				deb_print_pure_debug((uint8_t *)print_str, len);
+				THR_HARDCODED_SEQUENCES[procedure_id].execution_index++; // increase sequence execution index so that after wait - next module to be executed
+				THR_HARDCODED_SEQUENCES[procedure_id].substage_index = 0;
+
+				//*******assume that argv[5] is custom print message !!!!! WARNING I AM NOT SURE THAT THIS IS GOOD IDEA
+				len = strlen(argv[5]);
+				deb_print_pure_debug((uint8_t *)argv[5], len);
+				//// WARNING THIS BLOCK MIGHT BE NO GOOD *********
+			}
+
+
+
+
+		break;
+	case 1:
+		sprintf(print_str, "\n Sending READ [%d] Request %d\n",register_index, now_timestamp );
+		len = strlen(print_str);
+		deb_print_pure_debug((uint8_t *)print_str, len);
+
+		temp_argv[0]= "6";
+		//temp_argv[1]= "97"; // Reservoir Temperature
+
+		sprintf(argument1, "%d",register_index);
+		temp_argv[1]= argument1;
+
+		GeneralReadRequest(2, temp_argv);
+
+		THR_HARDCODED_SEQUENCES[procedure_id].substage_index ++;
+		THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage = (uint32_t)timGetSystime();
+
+		break;
+	case 2:
+
+		if ( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage) < wait_for_read_request_proccess ){ // Wait to proccess READ request
+						// do nothing
+					}
+		else{
+			//THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage = (uint32_t)timGetSystime(); //save execution finish time of wait substage
+
+			sprintf(print_str, "\nMonitored [%d] Value = %.6f \n", register_index,REGISTER_DATA[register_index]);
+			len = strlen(print_str);
+			deb_print_pure_debug((uint8_t *)print_str, len);
+			THR_HARDCODED_SEQUENCES[procedure_id].substage_index = 0;
+
+
+			}
+
+
+		break;
+
+	}
+
+
+
+
+
+
+}
+
+
+void thr_void(int argc, char *argv[]){
+	LAST_STARTED_MODULE = 1109;
+
+	/*
```

Comment:
This is just function that does nothing in a sequence.  I thought that ..... it might be handy someday :) But for now it is redundent

My logic was that there could be something like jump_sequence(uint8_t index) 

which jumps to a specific index at hardcoded sequence. Can be handy if we need to implement some loops within sequences. 

For example - while waiting for right conditions we loop through monitoring section of sequence.  

to develop this further and implement - need details from SYSTEM ENGINEERING.